uitools/cpp/Esri/ArcGISRuntime/Toolkit/BasemapGalleryController.cpp

```diff
@@ -508,16 +508,38 @@ namespace Toolkit {
    */
   void BasemapGalleryController::setCurrentBasemap(Basemap* basemap)
   {
-    if (basemap == m_currentBasemap)
-      return;
-
-    m_currentBasemap = basemap;
-    emit currentBasemapChanged();
+      auto connection = new QMetaObject::Connection();
+
+      auto apply = [basemap, this, connection](Error e){
+        if(e.isEmpty()){
+            if (basemap == m_currentBasemap)
+                  return;
+            if(!basemapMatchesCurrentSpatialReference(basemap)){
+                //force redraw for all the lsitview items.
+                emit m_gallery->dataChanged(m_gallery->index(0), m_gallery->index(std::max(m_gallery->rowCount() - 1, 0)));
+                return;
```

Comment:
❌  I'd maybe put `connection` in a `shared_ptr`. There's memory leaks in these early returns.